TimeOfFlight/src/TOFEstimators.cc

```diff
@@ -133,26 +130,26 @@ void TOFEstimators::processEvent(EVENT::LCEvent * evt){
             //define tof as an average time between two SET strips
             //if no SET hits found, tof alreasy is 0, just skip
             TrackerHit* hitSET = getSETHit(track);
+
             if ( hitSET != nullptr ){
-                const vector<LCObject*>& simHitsSET = navigatorSET.getRelatedToObjects( hitSET );
-                if ( simHitsSET.size() >= 2 ){
-                    //It must be always 2, but just in case...
-                    if (simHitsSET.size() > 2) streamlog_out(WARNING)<<"Found more than two SET strip hits! Writing TOF as an average of the first two elements in the array."<<std::endl;
-
-                    SimTrackerHit* simHitSETFront = static_cast <SimTrackerHit*>( simHitsSET[0] );
-                    SimTrackerHit* simHitSETBack = static_cast <SimTrackerHit*>( simHitsSET[1] );
-                    double timeFront = RandGauss::shoot(simHitSETFront->getTime(), _timeResolution);
-                    double timeBack = RandGauss::shoot(simHitSETBack->getTime(), _timeResolution);
-                        timeOfFlight = (timeFront + timeBack)/2.;
+                const std::vector<LCObject*>& rawObjects = hitSET->getRawHits();
+                if ( rawObjects.empty() ){
+                    streamlog_out(WARNING)<<"Found no raw SET strip hits, but space point is built!? Writing TOF as 0."<<std::endl;
                 }
-                else if (simHitsSET.size() == 1){
-                    streamlog_out(WARNING)<<"Found only one SET strip hit! Writing TOF from a single strip."<<std::endl;
-                    SimTrackerHit* simHitSET = static_cast <SimTrackerHit*>(simHitsSET[0]);
-                    timeOfFlight = RandGauss::shoot(simHitSET->getTime(), _timeResolution);
+                else if( rawObjects.size() == 1 ){
+                    streamlog_out(WARNING)<<"Found only one SET strip hit, but space point is built!? Writing TOF from a single strip."<<std::endl;
+                    TrackerHitPlane* rawHit = dynamic_cast <TrackerHitPlane*>(rawObjects[0]);
```

Comment:
```suggestion
                    auto* rawHit = dynamic_cast <TrackerHitPlane*>(rawObjects[0]);
```
Not really necessary to fix, but e.g. `clang-tidy` would flag this ad duplicating the type unnecessarily.